tests/test_datetime_utils.py

```diff
@@ -1,37 +1,111 @@
+        self.assertEqual(c_l_tomorrow.day, dt_tomorrow.day)
+        self.assertEqual(c_l_tomorrow.month, dt_tomorrow.month)
+        self.assertEqual(c_l_tomorrow.year, dt_tomorrow.year)
 
-    def test_4(self):
+    def test_year(self):
         self.assertNotEqual(year_begin(), None)
+        self.assertEqual(year_begin(), datetime.utcnow().replace(month=1, day=1, hour=0, minute=0, second=0, microsecond=0))
         self.assertNotEqual(year_end(), None)
+        self.assertEqual(year_end(), datetime.utcnow().replace(year=datetime.utcnow().year + 1, month=1, day=1, hour=0, minute=0, second=0, microsecond=0))
+
+    def test_month(self):
+        self.assertNotEqual(month_begin(), None)
+        self.assertEqual(month_begin(), datetime.utcnow().replace(day=1, hour=0, minute=0, second=0, microsecond=0))
+        self.assertNotEqual(month_end(), None)
+        self.assertEqual(month_end(), (today().replace(day=1) + timedelta(days=32)).replace(day=1))
+
+    def test_week(self):
+        self.assertNotEqual(week_begin(), None)
+        self.assertEqual(week_begin(), today() - timedelta(days=datetime.today().weekday()))
+        self.assertNotEqual(week_end(), None)
+        self.assertEqual(week_end(), (today() - timedelta(days=datetime.today().weekday())) + timedelta(days=7))
+
+    def test_day(self):
+        self.assertNotEqual(day_begin(), None)
+        self.assertEqual(day_begin(), today())
+        self.assertNotEqual(day_end(), None)
+        self.assertEqual(day_end(), tomorrow())
+
+
+    def test_hour(self):
+        self.assertNotEqual(hour_begin(), None)
+        self.assertEqual(hour_begin(), datetime.utcnow().replace(minute=0, second=0, microsecond=0))
+        self.assertNotEqual(hour_end(), None)
+        self.assertEqual(hour_end(), datetime.utcnow().replace(hour=datetime.utcnow().hour + 1, minute=0, second=0, microsecond=0))
+
+    def test_sunday(self):
+        self.assertNotEqual(sunday(), None)
+        self.assertEqual(sunday(), self._next_weekday(today(), 6))
+
+    def test_monday(self):
+        self.assertNotEqual(monday(), None)
+        self.assertEqual(monday(), self._next_weekday(today(), 0))
+
+    def test_tuesday(self):
+        self.assertNotEqual(tuesday(), None)
+        self.assertEqual(tuesday(), self._next_weekday(today(), 1))
+
+    def test_wednesday(self):
+        self.assertNotEqual(wednesday(), None)
+        self.assertEqual(wednesday(), self._next_weekday(today(), 2))
+
+    def test_thursday(self):
+        self.assertNotEqual(thursday(), None)
+        self.assertEqual(thursday(), self._next_weekday(today(), 3))
+
+    def test_friday(self):
+        self.assertNotEqual(friday(), None)
+        self.assertEqual(friday(), self._next_weekday(today(), 4))
+
+
+    def test_saturday(self):
+        self.assertNotEqual(saturday(), None)
+        self.assertEqual(saturday(), self._next_weekday(today(), 5))
+
+    def test_midnight(self):
+        self.assertNotEqual(midnight(), None)
```

Comment:
hours, seconds are 0?

Reply:
having some issues with midnight, its returning weird values, need some clarification with that